Thheinen/efs accesspoints

mapk-amazon opened the pull request:
"Clone" of https://github.com/aws/aws-parallelcluster/pull/6359

### Description of changes
Add support to specify EFS access points per issue https://github.com/aws/aws-parallelcluster/issues/2337

### Tests
* Integration tests included in the `develop.yaml` as well as unit tests.

thheinen committed:
Add EFS access point support (aws-parallelcluster#2337)

Signed-off-by: Thomas Heinen <t.heinen@reply.de>

mapk-amazon committed:
Added tests for EFS Access Points

thheinen committed:
Rename new setting, Add changelog, Fix style, fix some tests

Signed-off-by: Thomas Heinen <t.heinen@reply.de>

dreambeyondorange commented on cli/src/pcluster/schemas/cluster_schema.py:
```diff
@@ -321,6 +321,9 @@ class EfsSettingsSchema(BaseSchema):
     )
+    access_point_id = fields.Str(
+        validate=validate.Regexp(r"^fsap-[0-9a-z]{17}$"), metadata={"update_policy": UpdatePolicy.SUPPORTED}
+    )
```
Has this update support been tested?

mapk-amazon replied:
Has not been tested. To be one the safe side I marked it as `unsupported` now.

dreambeyondorange commented on cli/src/pcluster/validators/efs_validators.py:
```diff
@@ -18,10 +18,16 @@ class EfsMountOptionsValidator(Validator):
+                "An access point can only be specified when using an existing EFS file system. "
+                f"Please either remove the access point id {access_point_id} or provide the file system id for the access point",
+                FailureLevel.ERROR,
+            )
```
This probably needs to reference the `file_system_id` param

mapk-amazon replied:
Replaced the wrong `name` by the `file-system-id`.

dreambeyondorange commented on cli/tests/pcluster/validators/test_efs_validators.py:
```diff
@@ -42,8 +42,42 @@
     ],
 )
-def test_efs_mount_options_validator(encryption_in_transit, iam_authorization, expected_message):
+def test_efs_mount_options_validator(
```
This test would need to be updated to reflect the `file_system_id` param above.

mapk-amazon replied:
Replaced the wrong `name` by the `file-system-id`.

dreambeyondorange reviewed (changes requested):
Sorry for the delay.  I think this PR looks great.  Thanks for the contribution.  I have a couple minor comments and once resolved I can approve!

mapk-amazon committed:
Improved unit test for access point

dreambeyondorange commented:
Approved but you need to rebase

mapk-amazon committed:
Access point requires TLS
https://docs.aws.amazon.com/efs/latest/ug/efs-access-points.html#mount-with-access-point
Added unit test

mapk-amazon commented:
Rebased to the latest develop. Needed to rewrite the history a bit... Should now be up-to-date.

mapk-amazon committed:
used black formatter

mapk-amazon committed:
Updated snapshot to included AccessPointId

dreambeyondorange committed:
Merge branch 'develop' into thheinen/efs-accesspoints

mapk-amazon committed:
code linter fixes

mapk-amazon committed:
Fixed Code Linters..

mapk-amazon committed:
Ignoring one formatter as it is in conflict with another...

mapk-amazon committed:
Ignored Flake8 too-long-line
